samples/cpp/image_generation/benchmark_image_gen.cpp

```diff
@@ -106,15 +106,50 @@ inline void print_statistic(std::vector<ov::genai::ImageGenerationPerfMetrics>&
               << " ms, vae decoder infer avg time:" << vae_decoder_mean << " ms" << std::endl;
 }
 
+inline std::vector<std::string> device_string_to_triplet(const std::string& device_input) {
+    std::vector<std::string> devices;
+    std::istringstream stream(device_input);
+    std::string device;
+
+    // Split the device input string by commas
+    while (std::getline(stream, device, ',')) {
+        devices.push_back(device);
+    }
+
+    // Trim whitespace from each device name
+    for (auto& dev : devices) {
+        dev.erase(0, dev.find_first_not_of(" \t"));
+        dev.erase(dev.find_last_not_of(" \t") + 1);
+    }
+
+    // Ensure exactly three devices
+    if (devices.size() == 1) {
+        return {devices[0], devices[0], devices[0]};
+    } else if (devices.size() == 3) {
+        return devices;
+    } else {
+        throw std::invalid_argument("The device specified by -d/--device must be a single device (e.g. -d \"GPU\"), "
+                                    "or exactly 3 comma separated device names (e.g. -d \"CPU,NPU,GPU\")");
+    }
+}
+
 void text2image(cxxopts::ParseResult& result) {
     std::string prompt = result["prompt"].as<std::string>();
     const std::string models_path = result["model"].as<std::string>();
-    std::string device = result["device"].as<std::string>();
+    auto devices = device_string_to_triplet(result["device"].as<std::string>());
     size_t num_warmup = result["num_warmup"].as<size_t>();
     size_t num_iter = result["num_iter"].as<size_t>();
     const std::string output_dir = result["output_dir"].as<std::string>();
 
-    ov::genai::Text2ImagePipeline pipe(models_path, device);
+    ov::genai::Text2ImagePipeline pipe(models_path);
+    if (result["reshape"].as<bool>()) {
+        pipe.reshape(result["num_images_per_prompt"].as<size_t>(),
+                     result["height"].as<size_t>(),
+                     result["width"].as<size_t>(),
+                     pipe.get_generation_config().guidance_scale);
+    }
+    pipe.compile(devices[0], devices[1], devices[2]);
```

Comment:
I think the time consumption (like https://github.com/openvinotoolkit/openvino.genai/blob/0113d58e4604f8f8fbdc5d5e42c833d5c407180d/src/cpp/src/image_generation/text2image_pipeline.cpp#L36) should be counted in the pipeline's `compile` functiion, `Image2ImagePipeline` and `InpaintingPipeline` need the same operation. And the implementation of the `save_load_time` function should be changed to accumulate and assign values to `m_load_time_ms`

Reply:
Makes sense, please see latest commit.